### DISCUSSION TITLE
Initiating a recurring donation payment from GiveWP

### DISCUSSION THREAD

**Discussion by joshAppDev** [impress-org/givewp#7609] [Developer Q&A] [answered]
I'm writing a custom payment gateway add-on for GiveWP. It's currently able to create one-time donations as well as recurring donations. However, the payment gateway I'm dealing with does not generate the recurring payments automatically. It requires the merchant (meaning GiveWP in this case) to do that instead. My question is, what is the recommended way to do subscription payments in the custom add-on?

For the initial payment, I just do that immediately at the same time the subscription is created. This is done in the `createSubscription()` method of my `SubscriptionModule` subclass. For payments after that, what is the proper GiveWP methodology?

Should I use `wp_schedule_event()`? Or maybe `Give_Cron::add_daily_event()`?

**Comment by jonwaldstein** [comment, marked as answer]
Hey @joshAppDev,

GiveWP offloads all subscription responsibility to the Payment Gateway and listens for renewals.  If the gateway needed to work in the opposite direction, you would unfortunately have to figure out how to handle that yourself as we do not provide a way.

**Reply by JasonTheAdams** [reply]
As a note, @joshAppDev, it _is_ possible to do with custom work. GiveWP does come with the [Action Scheduler](https://actionscheduler.org/) package, so you could make use of that to schedule your renewals. Note, though, that jobs like this are not guaranteed to run at exact times in WP, so make sure you're aware of the challenges of this. At the time of renewal, you'd need to create the donation, connect it to the subscription, and charge it with the gateway.

This is the exact reason we've always avoided scheduling these things ourselves in GiveWP — we've even worked directly with gateways to get them to support managing subscriptions themselves before we integrated.

So yeah, you can do what you're looking to do, but it's a lot more work. If you have control over the server your websites is running on, you could certainly make this reliable. You'd just need to know what you're doing.

Best of luck!
